src/lib/device/physical_dev.hpp

```diff
@@ -382,6 +386,14 @@ class PhysicalDevChunk {
     void free_slot() { m_chunk_info->set_slot_allocated(false); }
     void update_end_of_chunk(uint64_t sz) { m_chunk_info->end_of_chunk_size = int64_cast(sz); }
     void update_start_offset(uint64_t start_offset) { m_chunk_info->update_start_offset(start_offset); }
+    void set_user_data(uint8_t* user_data, uint8_t user_data_size) {
+        HS_REL_ASSERT_LE(user_data_size, MAX_USER_DATA_SIZE_IN_CHUNK);
+        std::memcpy(m_chunk_info->user_data, user_data, user_data_size);
+        if (user_data_size < MAX_USER_DATA_SIZE_IN_CHUNK) {
```

Comment:
Since the situation when 'user_data_size' is equal or larger than MAX_USER_DATA_SIZE_IN_CHUNK is ready asserted in line 389, do we still have to add the branch here?

Reply:
asset just make sure user_data_size is **less or equal** to MAX_USER_DATA_SIZE_IN_CHUNK, so if user_data_size is less,  line 393 will fill the unfilled tail area to zero.